### DISCUSSION TITLE
Generations in Optimization Tools

### DISCUSSION THREAD

**Discussion by shasanc41** [Tunny-gh/Tunny_in_Open_Sea#249] [General]
I was wondering about something fundamental about the tool. Usually such optimization tools have generations and solutions. So far each generation they run some number of solutions and with each generation he learns and improves the result. However, in your tool, it seems like you are using a term called Trials and the optimization runs for the number of trials that I input. I want to know if you're applying any different theory here to optimize. Let's say if I run the optimization for 100 trials and then run again for 20 trials, would a 20 trial result be as optimum as 100 trial? 

Thanks,
Shasan

**Comment by hrntsm** [comment]
In optimization, the concept of generation is present in optimization methods classified as evolutionary algorithms.
In Tunny, this means NSGA and CMA-ES.

The reason why Tunny uses the term "number of trials" is that it supports Bayesian optimization, Quasi-Monte Carlo, and Random, in addition to Evolutionary Algorithms.
There is no concept of generation in those methods

The generation when GA is selected can be set by PopulationsSize in the image.
Once the number of PopulationSize attempts have been made, the next generation will move on.

<img src="https://github.com/hrntsm/Tunny/assets/23289252/f42ce591-ca4a-437a-abed-425e5e582e5d" width="300">

For example, set PopulationSize to 30. If the number of trials is set to 100, optimization will be terminated in the middle of the fourth generation.
If you run an additional 20 trials, optimization will continue as the fourth generation.

Although the documentation has not been updated much, an overview of TPE, a form of Bayesian optimization, is provided below.
Take a look if you are interested.
https://tunny-docs.deno.dev/docs/technical-info/tree-structured-parzen-estimator
